spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala

```diff
@@ -167,16 +167,21 @@ class TreeNNAccuracy[T: ClassTag]()(
 /**
  * Caculate the percentage that output's max probability index equals target
  */
-class Top1Accuracy[T](
+class Top1Accuracy[T: ClassTag](
   implicit ev: TensorNumeric[T])
   extends ValidationMethod[T] {
   override def apply(output: Activity, target: Activity):
   ValidationResult = {
     var correct = 0
     var count = 0
 
-    val _output = output.asInstanceOf[Tensor[T]]
     val _target = target.asInstanceOf[Tensor[T]]
+    val _output = if (output.toTensor[T].size().head != _target.size().head) {
```

Comment:
why should we change this?

Reply:
For dnn model, the model's output is consistent. For example, we have `32x3x224x224` input and `32x1000` as output. When the input's batch size is less than 32, the output will output `32x1000` also, but the target will be not. So we should narrow the useful elements.